src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -620,4 +521,81 @@ public int getNumOfMsqChoices() {
         return msqChoices;
     }
 
+    /**
+     * Getting number of responses.
+     * @return -1 if there is no empty response else number of response.
+     */
+    private int getNumberOfResponse(
+            List<FeedbackResponseAttributes> responses, Map<String, Integer> answerFrequency) {
+        boolean isContainsNonEmptyResponse = false; // we will only show stats if there is at least one nonempty response
+
+        for (String option : msqChoices) {
+            answerFrequency.put(option, 0);
+        }
+
+        if (otherEnabled) {
+            answerFrequency.put("Other", 0);
+        }
+
+        int numChoicesSelected = 0;
+        for (FeedbackResponseAttributes response : responses) {
+            List<String> answerStrings =
+                    ((FeedbackMsqResponseDetails) response.getResponseDetails()).getAnswerStrings();
+            boolean isOtherOptionAnswer =
+                    ((FeedbackMsqResponseDetails) response.getResponseDetails()).isOtherOptionAnswer();
+            String otherAnswer = "";
+
+            if (isOtherOptionAnswer) {
+                if (!answerFrequency.containsKey("Other")) {
+                    answerFrequency.put("Other", 0);
+                }
+
+                answerFrequency.put("Other", answerFrequency.get("Other") + 1);
+
+                numChoicesSelected++;
+                // remove other answer temporarily to calculate stats for other options
+                otherAnswer = answerStrings.get(answerStrings.size() - 1);
+                answerStrings.remove(otherAnswer);
+            }
+
+            int numEmptyChoicesSelected = getNumberOfEmptyResponseOfQuestion(answerStrings, answerFrequency);
+            if (numEmptyChoicesSelected > 0) {
+                isContainsNonEmptyResponse = true;
+                numChoicesSelected += numEmptyChoicesSelected;
+            }
+
+            // restore other answer if any
+            if (isOtherOptionAnswer) {
+                answerStrings.add(otherAnswer);
+            }
+        }
+
+        if (!isContainsNonEmptyResponse) {
+            return -1;
+        }
+
+        return numChoicesSelected;
+    }
+
+    private int getNumberOfEmptyResponseOfQuestion(List<String> answerStrings, Map<String, Integer> answerFrequency) {
```

Comment:
This method looks like it gets the number of non empty responses instead, considering it uses `continue` if the string is empty.

Also, it should use the plural form `responses` instead.